libs/ledger-state/src/Cardano/Ledger/State/Transform.hs

```diff
@@ -1,54 +1,87 @@
+      {-# UNPACK #-} !Word64 -- Payment Addr
+      {-# UNPACK #-} !Word64 -- Payment Addr (32bits) + ... +  0/1 for Testnet/Mainnet + 0/1 Script/Pubkey
+      {-# UNPACK #-} !(CompactForm Coin) -- Ada value
+  | TxOutStakingDH'
+      !(Credential 'Staking C)
+      {-# UNPACK #-} !Word64 -- Payment Addr
+      {-# UNPACK #-} !Word64 -- Payment Addr
+      {-# UNPACK #-} !Word64 -- Payment Addr
+      {-# UNPACK #-} !Word64 -- Payment Addr (32bits) + ... +  0/1 for Testnet/Mainnet + 0/1 Script/Pubkey
+      {-# UNPACK #-} !(CompactForm Coin) -- Ada value
+      {-# UNPACK #-} !Word64 -- DataHash
+      {-# UNPACK #-} !Word64 -- DataHash
+      {-# UNPACK #-} !Word64 -- DataHash
+      {-# UNPACK #-} !Word64 -- DataHash
 
 instance NFData TxOut' where
   rnf (TxOut' _ _) = ()
-  rnf (TxOutNoStake' _) = ()
+  rnf (TxOutDH' _ _ _) = ()
+  rnf TxOutStaking' {} = ()
+  rnf TxOutStakingDH' {} = ()
 
--- transTxOut :: TxOut CurrentEra -> TxOut' CurrentEra
--- transTxOut = \case
+toTxOut' :: Alonzo.TxOut CurrentEra -> TxOut'
+toTxOut' txOut =
+  case txOut of
+    Alonzo.TxOutCompact' cAddr cVal -> TxOut' cAddr cVal
+    Alonzo.TxOutCompactDH' cAddr cVal dh -> TxOutDH' cAddr cVal dh
+    Alonzo.TxOut_AddrHash28_AdaOnly cred a b c d ada ->
+      TxOutStaking' cred a b c d ada
+    TxOut_AddrHash28_AdaOnly_DataHash32 cred a b c d ada o p q r ->
+      TxOutStakingDH' cred a b c d ada o p q r
 
-toTxOut' :: Map (StakeCredential C) a -> Alonzo.TxOut CurrentEra -> TxOut'
-toTxOut' m txOut =
+toOrigTxOut' :: Alonzo.TxOut CurrentEra -> TxOut'
+toOrigTxOut' txOut =
   case txOut of
-    Alonzo.TxOutCompact cAddr cVal
-      | Just (cAddr', sr) <- restructureAddr cAddr ->
-        TxOut' (Alonzo.TxOutCompact cAddr' cVal) sr
-    Alonzo.TxOutCompactDH cAddr cVal dh
-      | Just (cAddr', sr) <- restructureAddr cAddr ->
-        TxOut' (Alonzo.TxOutCompactDH cAddr' cVal dh) sr
-    _ -> TxOutNoStake' txOut
-  where
-    restructureAddr cAddr =
-      case decompactAddr cAddr of
-        Addr ni pc (StakeRefBase sr) ->
-          Just (compactAddr (Addr ni pc StakeRefNull), intern sr m)
-        _ -> Nothing
+    Alonzo.TxOutCompact cAddr cVal -> TxOut' cAddr cVal
+    Alonzo.TxOutCompactDH cAddr cVal dh -> TxOutDH' cAddr cVal dh
+
+-- toTxOutWithSharing' :: Map (StakeCredential C) a -> Alonzo.TxOut CurrentEra -> TxOut' CurrentEra
+-- toTxOut' m txOut =
+--   case txOut of
+--     Alonzo.TxOutCompact cAddr cVal
+--       | Just (cAddr', sr) <- restructureAddr cAddr ->
+--         TxOut' (Alonzo.TxOutCompact cAddr' cVal) sr
+--     Alonzo.TxOutCompactDH cAddr cVal dh
+--       | Just (cAddr', sr) <- restructureAddr cAddr ->
+--         TxOut' (Alonzo.TxOutCompactDH cAddr' cVal dh) sr
+--     _ -> TxOutNoStake' txOut
+--   where
+--     restructureAddr cAddr =
+--       case decompactAddr cAddr of
+--         Addr ni pc (StakeRefBase sr) ->
+--           Just (compactAddr (Addr ni pc StakeRefNull), intern sr m)
+--         _ -> Nothing
```

Comment:
Dead code?

Reply:
You are right, it is no longer needed. I dumped most of the content in that module, since it was just experimentation with a better compaction of TxOut